app/Http/Controllers/AuthController.php

```diff
@@ -0,0 +1,102 @@
+<?php
+namespace App\Http\Controllers;
+use Illuminate\Http\Request;
+use App\User;
+use Socialite;
+//use Laravel\Socialite\Facades\Socialite;
+use Auth;
+class AuthController extends Controller
+{
+    //
+    public function redirectToFacebook()
+{
+    return Socialite::with('facebook')->redirect();
+}
+
+public function getFacebookCallback()
+{
+    $data = Socialite::with('facebook')->user();
+    $user = User::where('email', $data->email)->first();
+    
+    if(!is_null($user)) {
+        Auth::login($user);
```

Comment:
Rather then call `Auth::login` twice, we should just call it once at the end of the method.